libopae/src/reconf.c

```diff
@@ -289,11 +307,16 @@ fpga_result __FPGA_API__ fpgaReconfigureSlot(fpga_handle fpga,
 		goto out_unlock;
 	}
 
-	// Clear port errors
-	result = clear_port_errors(fpga);
-	if (result != FPGA_OK) {
-		FPGA_ERR("Failed to clear port errors.");
+	// Clear port errors for enough time for a max of 64 trasactions
+	int i;
+	for (i = 0; i < 2*MAX_TRANSACTIONS; ++i) {
+		result = clear_port_errors(fpga);
+		if (result != FPGA_OK) {
+			FPGA_ERR("Failed to clear port errors.");
+			goto out_unlock; // Why don't we exit here?
+		}
 	}
+	printf("Finished Clearing Errors");
```

Comment:
Please avoid using `printf()` in library functions. Users may not want to get the message in their output. Is this debug output (`FPGA_DBG()`), or is this something you want to relay as an intermediate state to the user?

Reply:
thanks for this suggestion